src/common/actions/auth-store.js

```diff
@@ -223,10 +224,11 @@ function checkSignedIntoStoreError(error) {
 }
 
 async function fetchAccountInfo(root, discharge) {
-  const query = { root, discharge };
-  const accountUrl = `${BASE_URL}/api/store/account?${qs.stringify(query)}`;
-  const response = await fetch(accountUrl, {
-    headers: { 'Accept': 'application/json' }
+  const response = await fetch(`${STORE_API_URL}/account`, {
+    headers: {
+      'Authorization': `Macaroon root="${root}", discharge="${discharge}"`,
```

Comment:
Hehe, same point raised again as in https://github.com/canonical-ols/build.snapcraft.io/pull/683 - same answer tl;dr: I don't think it's worth it.